### PR TITLE
Sanitization via DOMPurify

### PR DESCRIPTION
**Pull request by shockey**
### Description

This PR removes `sanitize-html` in favor of `dompurify`.

It also introduces `jsdom` to our unit testing environment, since DOMPurify needs a DOM in order to function.

### Motivation and Context

While analyzing Swagger-UI's bundle sizes with a new tool during a conversation with @webron, I noticed that `postcss` was taking up quite a chunk of our bundle (see below). 

Aware that the library wasn't doing anything useful for us, I set out to remove whatever was pulling it in (it was `sanitize-html`, with a candlestick, in the library!)

image

### How Has This Been Tested?

I updated the Markdown tests to reflect some inconsequential changes in what DOMPurify's output looks like.

All the XSS-related tests continued to pass.

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [ ] My changes can not or do not need to be tested.
- [x] My changes can and should be tested by unit and/or integration tests.
- [x] If yes to above: I have added tests to cover my changes.
- [x] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by shockey** [comment]
cc @srpalomino - per our discussion this week, you're likely to touch the Markdown components as well, just an fyi!

File: src/core/components/providers/markdown.jsx [hunk 1]
```diff
@@ -40,20 +40,8 @@ Markdown.propTypes = {
 
 export default Markdown
 
-const sanitizeOptions = {
```
Comment (heldersepu) [inline src/core/components/providers/markdown.jsx]: So all these "fancy" options we had are all covered by default on DomPurify?
I guess if all the Unit Test pass we are good, right?

Comment (shockey) [reply src/core/components/providers/markdown.jsx]: @heldersepu yeah, looks like this library tends to have a lighter touch 😄

**Comment by srpalomino** [comment]
Thanks for the heads up Shockey.

I’ll definitely keep this in mind!

**Pull request merged**